test/gtest/uct/ib/test_cqe_zipping.cc

```diff
@@ -0,0 +1,161 @@
+/**
+* Copyright (C) NVIDIA Corporation. 2022.  ALL RIGHTS RESERVED.
+*/
+
+#include <common/test.h>
+#include <common/test_helpers.h>
+#include <uct/ib/test_ib.h>
+
+extern "C" {
+#include <uct/api/uct.h>
+#include <uct/ib/base/ib_iface.h>
+}
+
+#define UCT_INSTANTIATE_MLX5_TEST_CASE(_test_case) \
+    _UCT_INSTANTIATE_TEST_CASE(_test_case, rc_mlx5) \
+    _UCT_INSTANTIATE_TEST_CASE(_test_case, dc_mlx5) \
+    _UCT_INSTANTIATE_TEST_CASE(_test_case, ud_mlx5)
+
+class test_cqe_zipping : public test_uct_ib_with_specific_port {
+public:
+    static bool check_counters()
+    {
+#if defined(ENABLE_STATS)
+        return !RUNNING_ON_VALGRIND;
+#else
+        return false;
+#endif
+    }
+
+    void send_while_possible()
+    {
+        while (am_zcopy() != UCS_ERR_NO_RESOURCE) {
+            ++m_send_cnt;
+        }
+    }
+
+    void wait_for_completion() const
+    {
+        /*
+         * Local deadline prevents hanging inside receive loop in case of
+         * connectivity issues
+         */
+        const ucs_time_t deadline = ucs::get_deadline(60);
+        while ((m_send_cnt != m_recv_cnt) && (ucs_get_time() < deadline)) {
+            progress();
+        }
+        ASSERT_EQ(m_send_cnt, m_recv_cnt);
+    }
+
+    size_t get_total_zipped_count() const
+    {
+#if defined(ENABLE_STATS)
+        constexpr int counters[2] = {UCT_IB_IFACE_STAT_TX_COMPLETION_ZIPPED,
+                                     UCT_IB_IFACE_STAT_RX_COMPLETION_ZIPPED};
+        size_t zipped_cqes_count  = 0;
+
+        for (auto &entity : m_entities) {
+            auto stats = ucs_derived_of(entity->iface(), uct_ib_iface_t)->stats;
+            for (auto &counter : counters) {
+                zipped_cqes_count += UCS_STATS_GET_COUNTER(stats, counter);
+            }
+        }
+        return zipped_cqes_count;
+#else
+        return 0;
+#endif
+    }
+
+    virtual void init()
+    {
+#if defined(ENABLE_STATS)
```

Comment:
is this if really needed?

Reply:
Yes, it is. Without this `ifdef` the test will fail with the following error:

```
Invalid UCS configuration for STATS_DEST : file:/dev/null, error message: No such element(-12)
```

I can prevent this error by moving `ifdef` inside `stats_activate` and do the same thing with the `stats_restore` function with regard to [the comment below](https://github.com/openucx/ucx/pull/8104#discussion_r862645562). But I prefer to leave both this calls external and implicit `ifdef` since it make the code more clear and straightforward from my point of view. And in sake of consistency with other use cases of this API of cource.

Reply:
can we move this inside stats_activate/restore functions?

Reply:
Decided to merge this request as is and cleanup this stuff in a separate patch.